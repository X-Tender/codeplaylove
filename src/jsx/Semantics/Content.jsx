import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	className: PropTypes.string,
	children: PropTypes.node,
};

const defaultProps = {
	className: '',
	children: null,
};

const Content = ({ className, children }) => {
	const classes = `content ${className}`;
	return <div className={classes}>{children}</div>;
};

Content.propTypes = propTypes;
Content.defaultProps = defaultProps;

export default Content;
