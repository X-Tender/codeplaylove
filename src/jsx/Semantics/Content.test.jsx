import React from 'react';
import { shallow } from 'enzyme';
import Content from './Content';

describe('Content', () => {
	const className = 'content';
	const children = <p>Hello there</p>;

	describe('with all props', () => {
		const contentWithAllProps = shallow(<Content className={className}>{children}</Content>);

		it('renders correctly', () => {
			expect(contentWithAllProps).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			const content = contentWithAllProps.find('div.content');

			expect(content.hasClass(className)).toBe(true);
			expect(content.contains(children)).toBe(true);
		});
	});

	describe('with no props', () => {
		const contentWithNoProps = shallow(<Content />);

		it('renders correctly', () => {
			expect(contentWithNoProps).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			const content = contentWithNoProps.find('div.content');
			expect(content.children()).toHaveLength(0);
		});
	});
});
