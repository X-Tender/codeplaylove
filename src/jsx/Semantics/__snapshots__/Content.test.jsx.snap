// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content with all props renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    className="content"
  >
    <p>
      Hello there
    </p>
  </Content>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Hello there
      </p>,
      "className": "content content",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello there",
      },
      "ref": null,
      "rendered": "Hello there",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Hello there
        </p>,
        "className": "content content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello there",
        },
        "ref": null,
        "rendered": "Hello there",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Content with no props renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    className=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "className": "content ",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "content ",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
